City-of-Helsinki/hel-fi-drupal-grants
# fix: AU-1612, AU-1672: End date restriction logic

# [AU-1612](https://helsinkisolutionoffice.atlassian.net/browse/AU-1612), [AU-1672](https://helsinkisolutionoffice.atlassian.net/browse/AU-1672)

## What was done
This pull request implements end date restriction logic on various grant application form date fields. Some of the forms have date field pairs on them containing a start date and an end date. The restriction logic makes it so that the end date can never come before the start date, which it previously could.

### The affected grants applications and their date fields are the following:

- **Liikunta: Tapahtuma-avustushakemus**
  - `Alkaa` and `Päättyy` fields.
  
- **Nuorisotoimi, projektiavustushakemus**
  - `Projekti alkaa` and `Projekti loppuu` fields.
  
- **Kasvatus ja koulutus: Iltapäivätoiminnan harkinnanvarainen lisäavustushakemus**
  - `Alkaen` and `Päättyy` fields.
  
- **Taide- ja kulttuuriavustukset: projektiavustukset**
  - `Hanke alkaa` and `Hanke loppuu` fields.
  
- **Taide ja kulttuuri: kehittämisavustus**
  - `Hanke alkaa` and `Hanke loppuu` fields.

## How to install

Make sure your instance is up and running on the correct branch.
  * `git checkout fix/AU-1612-form-date-fix`
  * `make fresh`
  * `make drush-cr`

## How to test
- Login with a test user and select the registered community role. Test all the listed forms above, and make sure that date fields work in the following way:
  - The end date must come after the start date.
  - If the start date value is changed to a value that is after the end date, then the end date field value should change so that it matches the value of the start date.
- Review the code.


[AU-1612]: https://helsinkisolutionoffice.atlassian.net/browse/AU-1612?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[AU-1672]: https://helsinkisolutionoffice.atlassian.net/browse/AU-1672?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
